network/tests/test_linux.py

```diff
@@ -0,0 +1,338 @@
+    'system.net.ip.in_csum_errors.count': 0,
+    'system.net.ip.reassembly_overlaps.count': 0,
+}
+
+
+CONNTRACK_STATS = {
+    'system.net.conntrack.found': (27644, 21960),
+    'system.net.conntrack.invalid': (19060, 17288),
+    'system.net.conntrack.ignore': (485633411, 475938848),
+    'system.net.conntrack.insert': (0, 0),
+    'system.net.conntrack.insert_failed': (1, 1),
+    'system.net.conntrack.drop': (1, 1),
+    'system.net.conntrack.early_drop': (0, 0),
+    'system.net.conntrack.error': (0, 0),
+    'system.net.conntrack.search_restart': (39936711, 36983181),
+}
+
+
+CONNECTION_QUEUES_METRICS = ['system.net.tcp.recv_q', 'system.net.tcp.send_q']
+
+
+def os_list_dir_mock(location):
+    if location == '/sys/class/net':
+        return ['ens5', 'lo', 'invalid']
+    elif location == '/sys/class/net/lo/queues':
+        return ['rx-1', 'tx-1']
+    elif location == '/sys/class/net/ens5/queues':
+        return ['rx-1', 'rx-2', 'tx-1', 'tx-2', 'tx-3']
+    elif location == '/sys/class/net/invalid/queues':
+        raise OSError()
+
+
+def ss_subprocess_mock(*args, **kwargs):
+    if '--udp --all --ipv4' in args[0][2]:
+        return '3', None, None
+    elif '--udp --all --ipv6' in args[0][2]:
+        return '4', None, None
+    elif '--tcp --all --ipv4' in args[0][2]:
+        file_name = 'ss_ipv4_tcp_short'
+    elif '--tcp --all --ipv6' in args[0][2]:
+        file_name = 'ss_ipv6_tcp_short'
+
+    with open(os.path.join(FIXTURE_DIR, file_name), 'rb') as f:
+        contents = f.read()
+        return decode_string(contents), None, None
+
+
+def ss_subprocess_mock_fails(*args, **kwargs):
+    if args[0][2].startswith('ss '):
+        raise OSError('boom')
+    else:
+        return get_subprocess_output(*args, **kwargs)
+
+
+def netstat_subprocess_mock(*args, **kwargs):
+    if args[0][0] == 'sh':
+        raise OSError()
+    elif args[0][0] == 'netstat':
+        with open(os.path.join(FIXTURE_DIR, 'netstat'), 'rb') as f:
+            contents = f.read()
+            return decode_string(contents), None, None
+
+
+def read_int_file_mock(location):
+    if location == '/sys/class/net/lo/mtu':
+        return 65536
+    elif location == '/sys/class/net/lo/tx_queue_len':
+        return 1000
+    elif location == '/sys/class/net/ens5/mtu':
+        return 9001
+    elif location == '/sys/class/net/ens5/tx_queue_len':
+        return 1000
+    elif location == '/sys/class/net/invalid/mtu':
+        return None
+    elif location == '/sys/class/net/invalid/tx_queue_len':
+        return None
+
+
+@mock.patch('datadog_checks.network.network.Platform.is_linux', return_value=True)
+def test_returns_the_right_instance(is_linux, check):
+    check_instance = check({})
+    assert isinstance(check_instance, LinuxNetwork)
+
+
+@pytest.mark.skipif(not Platform.is_linux(), reason="Only works on Linux systems")
+def test_collect_cx_queues(check, aggregator):
+    instance = copy.deepcopy(common.INSTANCE)
```

Comment:
I've been wondering what the difference is between this and providing `INSTANCE` as a per-test fixture. I'm guessing performance isn't an issue here, right?

Reply:
No difference, but since it's a constant I don't really see the need of using a fixture

Reply:
When we use an `instance` fixture it's typically the same thing, we simply abstract the `copy.deepcopy` out inside the fixture.